EngineBase.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace BooruDownloader
+{
+    public abstract class EngineBase
```

Comment:
Create a generic class for both of the engines, since they have the same functions.